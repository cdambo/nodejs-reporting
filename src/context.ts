export default class Context {
  private readonly globalTags: object;

  public constructor(globalTags: object = {}) {
    this.globalTags = globalTags;
  }

  public mergeTags(tags: object = {}): object {
    return Object.assign({}, this.globalTags, tags);
  }
}
