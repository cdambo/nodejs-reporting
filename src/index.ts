import Context from "./context";

export * from "./metrics";
export { Context };
