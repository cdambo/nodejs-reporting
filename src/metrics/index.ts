export * from "./formats";
export * from "./reporters";
export * from "./integrations";
