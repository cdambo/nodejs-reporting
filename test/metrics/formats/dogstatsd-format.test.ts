import { DogStatsDFormat } from "../../../src";

describe("DogStatsDFormat", (): void => {
  describe("format", (): void => {
    it("Formats tags to a comma separated list of snake case tags", (): void => {
      expect(
        DogStatsDFormat.format({
          stat: "metric_name",
          value: 17,
          sampleRate: 0.8,
          tags: {
            camelCaseTag1: "tagOne",
            camelCaseTag2: "tag_two",
            anotherTag: 5
          }
        })
      ).toMatchSnapshot();
    });
  });
});
