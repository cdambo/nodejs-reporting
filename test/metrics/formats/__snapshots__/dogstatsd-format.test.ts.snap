// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogStatsDFormat format Formats tags to a comma separated list of snake case tags 1`] = `
Object {
  "sampleRate": 0.8,
  "stat": "metric_name",
  "tags": Array [
    "camel_case_tag_1:tagOne",
    "camel_case_tag_2:tag_two",
    "another_tag:5",
  ],
  "value": 17,
}
`;
